### DISCUSSION TITLE
[Game Economy Discussion] Resource Burning

### DISCUSSION THREAD

**Discussion by TheDcol** [sunflower-land/sunflower-land#2379] [General]
_A discussion on how the game is currently handling burning/sink of resources and evaluating whether it is sustainable in the longterm. TLDR at the very bottom_

There have been a few topics discussing this in general, I would like to open up a discussion specifically on how the game encourages/forces/wants the player to burn their resources. I have talked about it before and other people have done so as well, although, this is purely in regards to burning.

Essentially, Sunflower Land is a game where players are able to gather resources over a period of time. Players buy seeds/tools for X cost and are then able to sell what they gather for X+. However much players spend, they will always return a profit via shop or trader after spending time growing/gathering resources.

The ONLY mandatory costs a player has to worry about are:
- Syncing/Restocking
- Trader Fees
- Withdraw Fees

### **HOW THE GAME BURNS RESOURCES**
The game offers players a chance to mint nfts, buildings or other items to further increase their productivity on their farm. Players often have to offer many resources in return for these items.. it is here where we see most of the action happen in Sunflower Land, we saw this recently with Treasure Island and the War Event.
Increased trade activity, interest and determination from players to try and acquire these items. Game Events/Special Items generally try to encourage resource burning as a way to combat the massive wave of resources generated by players everyday.

This is all good and well, I love events. But I think there is an innate issue in-between these special mints and events where the core of the game is exposed. Without events and new mints, we really are just generating resources non-stop , our only costs being syncing or withdraw fees, which do not really feel like fun game mechanics and are usually done because the player is profiting in the first place. Is this healthy? Is this okay? **Are events/nft releases the best and only way for the game to burn resources?**

### **THE ISSUE I SEE**
What happens when there are no new items released or no events running for a while? Is it a healthy burn system one that devs have to constantly make and release new items/events to apply ANY burning to resources? These events/mints are ALL optional so are resources even being burnt in the first place if only 1-5k players get involved while the other 15-20k do not? After all the events and special mints, Treasure Island and whatnot, they do not really help long-term:
![image](https://user-images.githubusercontent.com/108110065/227820697-8cd54f78-d88a-47f1-9aa0-d9d3b1865f4d.png)
![image](https://user-images.githubusercontent.com/108110065/227820706-0808c2f4-1086-40a0-abe5-5cd4d3881fb0.png)
![image](https://user-images.githubusercontent.com/108110065/227820711-5a4e0e20-81ea-4a76-b81b-586ff2718edb.png)

### **SOLUTION?**
Maybe we do not need a solution, maybe 'it's just how it is' and it's how the game is played out. Resources continually inflating, farmers producing more and more of everything, expanding more and minting nfts that allow them to produce more than before. To me it is a problem in the longterm, it's too much of an snowball effect. Players gaining skills, nfts, getting bigger and producing more for less and less of a cost. Game economies are very complex, especially ones that have a lot of players involved AND ones that are tied with crypto, attracting a lot of players KEEN to get their $0.5-$5 everyday...

I think there is space in the game to add lots of interesting innate burn mechanics.. ones that players cannot skip or ignore. Games often provide players with challenges, and often times part of the fun is overcoming these challenges (at least for me, as a gamer) ESPECIALLY IF incorporated well within the game itself.

Part of the solution would be to add scaling costs to farms and to buildings within. The bigger your farm, the more costs. I will not go into detail here as I think it deserves an entirely different thread, I have been working on a 'Farm Maintenance' proposal which will prove controversial but I think necessary to add a bit more interest and challenge to the game. More importantly, it aims at curbing the issue of resource inflation AND add interest/activity amongst players. "But Dcol wtf y u want to add more costs to the game?? i wanna make profit" Good luck making PrOFit when there are a million golds in the market and players leave because the game is repetitive by nature without constant bandaid events and mints :D

Some very short/summarised examples of farm maintenance mechanics:
- A farm with 3 expansions might have to burn 2 axes because a tree has fallen onto the road blocking access to goblin trader. A farm with 20 expansions might have to pay 20-30 axes. Not clearing the road = no access to goblin trader.
- All food buildings now require 1, 5 or 20 wood a day (depending on farm size) to function properly. More wood added = faster cooking time.
- Risk of rat infestation eating away at your stored crops, need to buy rat poison via npcs by trading minerals, fruit or other (Could incorporate interplay within economies, something supersynapse made a post about). The smaller the farm, the smaller the cost.

Seasons could also become part of the game, every 1 -3 months seasons could change and provide debuffs/buffs to certain things:
- Drought season meaning crops take longer to grow (Hello gnomes?), or lack of water means water wells start to deplete and need to refill it 
- Winter season means wheat/kale/radish take 3x to grow BUT also sell for a lot more in the shop

_Players will still want to expand as much as possible to unlock special crops, minerals and to be able to generate more space for more buildings. Importantly, the cost cannot be so high that it deters players from expanding._

**### SUMMARY/TLDR**
Without events or special mints, this game IMO lacks proper burn mechanics and just encourages players to generate resources non-stop.  Not only does it lead to resource inflation but I think it results in very repetitive boring gameplay.
This is a casual idle farming game, but that does not mean you cannot introduce some challenges in the game that could not only make it more interesting/fun to overcome, but it might help to create a sustainable economy without the constant need of events/special mints being added to the game. Special events/mints should just be that, special. Not absolute necessities to maintain interest and action within the market.

Thank you, hope you join the discussion =) Interested to hear some thoughts from people around this topic!

**Comment by Alaindrea** [comment]
I honestly don't see an issue here. I'm personally not trying to play a game like a financial tool, and I'm not extracting .05-$5/day, so I really don't care if there are a lot of carrots available for me to buy cheaply when I want carrots to craft the next event item. I'm honestly more concerned that the game seems very biased against letting me save up resources for the future or special events, as I would do on a real farm in case of a bad year. I don't think it is reasonable that players should ever *have* to buy resources to earn  event items without owning NFTs that new players can only buy on secondary. Furthermore, a player shouldn't be expected to play for 24hrs a day to earn them either. 

Look, Farmville was super dang popular and made a ton of money without ever penalizing you for saving the things you grow or harvest. 

If this game wants to continue to attract mainstream players to play the game, then staying active in the present needs to be the determining factor in getting items and such. this gives newer people a fair chance and encourages people to play more frequently instead of amassing humongous hordes of things because they have a broken beta nft.

**Reply by TheDcol** [reply]
> I'm honestly more concerned that the game seems very biased against letting me save up resources for the future or special events, as I would do on a real farm in case of a bad year.

The game currently allows for this, hoarding resources until the end of days. It's how I play too, I try and hoard as much as I can until it's needed for a future mint or event. Then the event/mint comes along, i use up the 5-10k resources, and I am back to hoarding.

It's a repetitive cycle imo. Farm, hoard, burn during event, farm hoard etc.
Which, maybe it's what majority of people want? Not sure, personally I would like to think a bit more, but could be just me.

> I'm personally not trying to play a game like a financial tool, and I'm not extracting .05-$5/day, so I really don't care if there are a lot of carrots available for me to buy cheaply when I want carrots to craft the next event item

That's great! Buying resources cheaper is good too, but do you think it would be more exciting in the game if the swings/demand/supply were more erratic and not just rely on Easter event once a year or other types of mints for carrots?

> Look, Farmville was super dang popular and made a ton of money without ever penalizing you for saving the things you grow or harvest.

Agreed. But Farmville was also a single player game, makes economy a lot less important. Being embedded in Facebook during its massive rise also had a part in their success aha

**Reply by wersout** [reply]
I understand what you mean but you can't compare farm ville to this game, every crop in sfl is a token that has a value so it have an impact on the whole econamy if hoarded or to much get burned... Have a good balance is very important with a good burn rate..
Game has same mecanics like real life and if every one hoarded everything nothing would have a value... 
Farm ville doesn't have to think of any of this it doesn't matter to them or econamy if you have 1 or 1 billion it doesn't effect game econamy but it does in this game like in real life so it's very important to have a good balance between hoard and burn and it needs to be of an big importans it works..
Devs want to build a web3 game and web3 is the future so going back to web2 like farm ville would be going back in development and not discover all the possibilites web3 gives....
But i understand how you think and that would be easy to go back to web2 but this game is developed to be a web3 game so econamy is very important...

**Reply by Alaindrea** [reply]
In theory it lets you, yes, but practically speaking if you don't have any of the OP NFTs ( I have nothing higher than T2 beaver/crops) then there is very little to amass. I end up with possibly 500 for the next event and I'm nearly maxed. that's a significant discrepancy

**Comment by brian-man** [comment]
> Risk of rat infestation eating away at your stored crops

Cat owners would love to see this arrive.

**Comment by WTFpacman** [comment]
Just a few Ideas from my side:
add Water as kinda semi ressource you use for growing crops. At a certain point you will need to "dig deeper" on all your water wells (each well could have his own capacity) with a new shovles that is mainly using wood and a few stone (maybe simalr scale then TI).

add wood burn to use every food building 2 options:
1. a weekly x burn of wood after it you can use the building as you want for 7 days.
2. a x wood/ time cooked burn that needs to be refilled when the storage runs low.

(it might be better to highly "nerf" this feature on the fire pit or just skip it so its still possible to use for new players)

add food burn to npc as "upkeep" every bumpkin needs food to work atleast daily.

**Comment by dragonhunter2017** [comment]
In my opinion, the best way to burn resources are to let people expand more with a significantly higher bumpkin level + resource requirements.. So,  players will continue to burn to expand  (that is the purpose of the game LE). 
There is so many clicking at the moment. Now if we had to maintain our building to operate, it will not be welcomed.

**Reply by TheDcol** [reply]
> There is so many clicking at the moment. Now if we had to maintain our building to operate, it will not be welcomed.

Yes there is a lot of repetitive clicking, clicking plots trees etc. This is not 'fun' I agree!

It's not fun because there is no thinking or challenge involved

**Comment by Chucrutty** [comment]
![image](https://user-images.githubusercontent.com/125780525/228352531-68677618-acf7-4f96-b6d1-66cdf942561c.png)

![image](https://user-images.githubusercontent.com/125780525/228352606-25e19c6f-a161-4bc6-8a1c-d06eeb4aadbe.png)

![image](https://user-images.githubusercontent.com/125780525/228353630-7418e027-1f4a-41b0-959f-145472747221.png)

![image](https://user-images.githubusercontent.com/125780525/228352810-ceb4cc5d-cf45-4e85-a4dd-51062bf7493a.png)

![image](https://user-images.githubusercontent.com/125780525/228352951-bb5d4557-4390-4ba8-b7b0-c66fa3804889.png)

**Comment by celpower** [comment]
"What happens when there are no new items released or no events running for a while? " We take a freaking break? I don't want to play a game where I have to constantly wake up in the middle of the night to "try" to compete for a NFT. One thing I love about solar fare season is that you can only mint one of the top 2 NFTs. Unless you were given advantage by Devs mistake or really have no life or has the most killer NFTs like Wings/Foreman/Kubiko. If the game starts to feel more or a constant chore and expense, I personally would quit. This coming from someone who has a very tiny pool stake and never withdrawal any SFL from the game but has spent plenty of MATIC and SFL to keep playing as is.

**Reply by TheDcol** [reply]
> We take a freaking break? I don't want to play a game where I have to constantly wake up in the middle of the night to "try" to compete for a NFT

My point was whether we should have short-term intense burns (the ones you hate waking up in the middle of the night from) vs long-term intrinsic burns.

Sounds like you would be a fan of the latter

**Comment by drumdrumdrum3337** [comment]
Good thread, I hope it's fine to add here related comments from another discussion and a proposal, so that it's all in one place:

-Proposal - Public spaces for Isles -https://github.com/sunflower-land/sunflower-land/discussions/2197
- Comment related to internal burning mechanics:

1. Boat repairs - after a certain bumpkin level the boat we use for travel starts leaking water and breaking up overall, hence it needs to be repaired with wood, beeswax(once bees are live), and some sfl for goblins. The higher the level and the more travel destinations are available, the more maintenance is required. The only destinations that are always available would be Helios and Goblin Village, since they could be reached with primitive raft.
Example: after level 15, if not repaired, the boat is not able to travel as far as Treasure Island. Once bumpkin is at level 20, the boat need to be repaired again, and also upgraded a bit in order to reach Stone Haven. At level 24 repairs are needed again, otherwise Stone Haven becomes out of reach, and if boat is not repaired till level 27, then even more resources would be required, because now even Treasure Island is not reachable. And so on.
Note: eventually it might be a good idea to built a repairs shop where goblins would provide maintenance for wheelbarrows, tractors, sprinklers, etc.. The repairs shop could work as a sink.

2. Chicken manure as an ingredient for fertilizers for some crops. Not exactly cyclical, but adds some more utility to chickens. In real world, if prepared correctly, chicken manure is an excellent fertilizer for tomatoes, beans, pumpkins and fruit trees.

3. Chickens are not eternal - there should be a life cycle of the chicken. For example, if not fed, a chicken dies in 5 days. If it is fed, then it lays eggs for let's say 4 months, then dies. However, when it's between 2 and 3 months old, there's an option to use its meat for cooking - this would lead to less eggs produced, but good exp gained.

**Comment by poteitouweb3** [comment]
I believe that we should base ourselves more on a real-life economy and increase various depreciation mechanisms in the game. What happens is that everything has eternal duration, benefits and positive effects for life and this decreases the strategic need of the players.

### **Build Depreciation**

Even inanimate things have a life cycle. Buildings should also follow the same path by way of maintenance demands for their repair.

- **Build health percentage:** A newly placed build would have 100%. As the player uses it, this percentage begins to drop, and from 70% the output will be affected. 

In the bakery for example, a cake that would give 2500 exp would give 10% less (2250 exp). For the most hardcore players this would be the time to repair to regain 100% and therefore the maximum efficiency of the build but if you lack resources you will be well aware that you will have less and less return from your build. If the player repairs at 70% he will be able to continue "eternally" with the build but each maintenance done below 70% will consume a "life" of the build. When it consumes all the "lives" or when it reaches 0% it would break and be "burned" and then the player would need to build a new one.

The other buildings would have similar mechanisms. For tools, crafting would demand more resources to be done.

### **Further ideas**

- **Cooking requirements:** As @TheDcol well suggested, it would be ideal if the buildings for cooking demand wood and perhaps coal to cook the food. I would also increase the idea of a % fire efficiency. The game would tell you the minimum wood to cook that food and an extra value that if placed will give you a time bonus. Unlike NFTs this mechanism would require constant resource burning. E.g. "Minimum requirement: 10 wood - 5hrs 30min" / "Recommended requirement: 20 wood - 4hrs 45min

- **Risky quests (rng, betting, luck...):** Why not send our bumpkins on risk missions? These missions would work similar to Treasury Island. You would bet 'X' amount of resources for a small chance to get a rewarding prize! Statistically the amount of resources burned would be greater than the amount of resources distributed which will help a lot in controlling the inflation of the game. Also the team can weekly analyze which resource is inflating the most and demand it in missions to demand even more inflation control.

- **Incentive for holders:** The creation of an in-game staking system would also be a great way to stimulate some accumulation in exchange for constant rewards. E.g. staking massive amounts of resources, like 50k wood to receive 10 $SFL daily, or something inverse like staking 10k $SFL to receive some resource randomly every day.

**Comment by goingsnginx** [comment]
Perhaps Treasure Island could be better optimized from a game theory perspective, but the ultimate result is that only a portion of players will participate in scenarios that attempt to consume resources, while another part of players will always play the game with the goal of maximizing their own benefits. They will avoid any potential losses, so attempting to increase their resource consumption is difficult.I can imagine that if only consumption is increased, they will only complain about the decrease in income. In fact, it is impossible to take care of all players, and there is a natural contradiction.

If the gameplay has been optimized to the extreme and still cannot solve the inevitable problem of inflation, introducing some mild financial methods may be an option.

The worst result of inflation is a long-term decline in the token price, which can lead to the loss of confidence of large SFL holders. No one wants to buy someone else's profits. When large holders sell a large amount of SFL, the remaining players' earnings will decrease significantly, and ultimately fall into a death spiral.

Therefore, the ideal result is that the SFL token can bring a positive feedback loop without halving. The token continues to rise slightly under restraint, players' income increases, large holders have confidence, bringing a continuous influx of new players.

Personally, I think to achieve a positive feedback loop, an SFL staking function can be added with an APY of 3% -10%. Different APYs can be experienced based on the depth (level) of the game. This approach has the following advantages:

1.Encouraging people to participate deeply in the game.
2.More people choose to hold tokens rather than sell them immediately, reducing sell orders on the market.
3.Accelerating halving speed. Essentially, short-term inflation is exchanged for long-term deflation, so the impact of additional income on inflation should not be a concern.

Different viewpoints are welcome for discussion.

**Reply by TheDcol** [reply]
> If the gameplay has been optimized to the extreme and still cannot solve the inevitable problem of inflation, introducing some mild financial methods may be an option.If the gameplay has been optimized to the extreme and still cannot solve the inevitable problem of inflation, introducing some mild financial methods may be an option.

There is the problem of inflation affecting macro and then the problem of what to do with 100k potatoes or 5k eggs as a player.
I would like to see more challenges in the game and I want to struggle in a way trying to maintain my big farm. All we are doing is gathering resources to either sell at the shop or mint stuff

**Reply by goingsnginx** [reply]
It is true that the game needs to increase resource consumption scenes as much as possible. However, I feel that this alone is not enough to solve the problem of inflation in the entire economic system. I have studied a PC RPG game with a lifespan of up to 20 years. They have a large number of random consumption scenes and also suffered from inflation a few years ago when PC games were in decline. Currently, they rely on constantly attracting new players to alleviate the problem.

However, I found that the more serious problem is that it is difficult to find in web2 games where more than 50% of players have the primary goal of making money. But in the web3 world, it may be inevitable that more than 50% of players have the primary goal of making money. That is to say, random consumption scenes are much more effective in web2 than in web3. This makes me feel that even if web3 games achieve the limit in terms of gameplay, they cannot fundamentally affect macro inflation, and cannot achieve a positive cycle. However, compared to web2 games, the advantage of web3 games is that they can add some financial means like the Fed raising interest rates to control USD inflation in the real world. Of course, I also firmly believe that web3 games must mainly rely on gameplay to attract users, and finance is only an auxiliary means.

In fact, if the ladder staking function I mentioned is added, on the one hand, as the halving decreases more, the importance of staking increases, which has a better effect on controlling liquidity. The second biggest benefit is that the price of SFL has stronger support because stakers can earn long-term profits, which can offset the losses caused by those players who make money off of them, rather than just buying SFL to pay for others' profits as is currently the case. This approach only accelerates the halving process and almost no one suffers any pain. I really think that the combination of halving and staking may be a solution.

**Comment by DominicTing** [comment]
We may need a pvp system to increase game fun and increase resource burning.

**Reply by Chucrutty** [reply]
Dungeons.... Guilds..... missions..... boss.....
